sycl/include/sycl/access/access.hpp

```diff
@@ -54,7 +55,12 @@ enum class address_space : int {
   generic_space = 6, // TODO generic_space address space is not supported yet
 };
 
-enum class decorated : int { no = 0, yes = 1, legacy = 2 };
+enum class decorated : int {
+  no = 0,
+  yes = 1,
+  legacy __SYCL2020_DEPRECATED(
```

Comment:
The interface behind is: `If the value of access::decorated is access::decorated::legacy, the 1.2.1 interface is exposed. This interface is deprecated.`

IMO, that's probably the best place to notify this.

Reply:
My concern is that as a user, I can write the following code:

```c++
auto decoration = sycl::access::decorated::legacy;
if (decoration == sycl::access::decorated::no) {
  // ...
} else {
  // ...
}
```

And for it I expect to see no deprecation warnings in that case, because the enumerator itself is not deprecated by the spec.

I don't want to block this PR, but I would like us to be aligned with the spec. If we think that that legacy enumerator should not be used at all, can we then propose to mark it deprecated in the spec itself?

Reply:
I opened an issue for deprecating `legacy` (https://github.com/KhronosGroup/SYCL-Docs/issues/293) but there are concerns. As such I think it is best to remove the deprecation here. Sadly deprecation of partial specializations has some odd behavior in clang, so I've added a note for adding deprecation warnings whenever we can.